### PR TITLE
GCE: Allow updating the instance group template

### PR DESCRIPTION
**Pull request by justinsb**
We follow the same pattern as on AWS, where we create a new template,
because the templates are immutable.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/2426)

### CODE REVIEW

File: upup/pkg/fi/cloudup/gcetasks/instancetemplate.go [hunk 1]
```diff
@@ -47,88 +53,115 @@ type InstanceTemplate struct {
...
-	//			actual.Image = fi.String(imageURL.Project + "/" + imageURL.Name)
-	//		}
-	//	}
-	//}
-
-	if p.Metadata != nil {
-		actual.Metadata = make(map[string]*fi.ResourceHolder)
-		for _, meta := range p.Metadata.Items {
-			actual.Metadata[meta.Key] = fi.WrapResource(fi.NewStringResource(meta.Value))
+		//for i, disk := range p.Disks {
```
Comment (chrislovecnm) [inline upup/pkg/fi/cloudup/gcetasks/instancetemplate.go]: Can we leave a comment on why we have commented code here?

**Comment by justinsb** [comment]
Added comment about GCE local disks

**Comment by justinsb** [comment]
ping @chrislovecnm

**Comment by justinsb** [comment]
Rebased & squashed for cherry-picking

File: upup/pkg/fi/cloudup/utils.go [hunk 2]
```diff
@@ -55,7 +55,7 @@ func BuildCloud(cluster *api.Cluster) (fi.Cloud, error) {
 
 			project = cluster.Spec.Project
 			if project == "" {
-				return nil, fmt.Errorf("project is required for GCE")
+				return nil, fmt.Errorf("project is required for GCE - try gcloud config get-value project")
```
Comment (geojaz) [inline upup/pkg/fi/cloudup/utils.go]: 👍

**Pull request merged**